module_utils/network_lsr/argument_validator.py

```diff
@@ -373,6 +374,16 @@ def __init__(self, name, family=None, required=False, default_value=None):
                 ArgValidatorNum(
                     "metric", default_value=-1, val_min=-1, val_max=0xFFFFFFFF
                 ),
+                ArgValidatorNum(
+                    "table", default_value=None, val_min=1, val_max=253
+                ),
+                ArgValidatorStr(
+                    "dev", default_value=None, enum_values=listdir('/sys/class/net/')
```

Comment:
Thank you very much for your PR. This is just some initial feedback that I saw from glancing over it. It seems to me that these `enum_values` are not correct. They would not allow adding rules for a virtual interface that is created in the same ansible run. This applies also to the other instances.

Reply:
Do you think I should remove this or try fetching values of those possible new interfaces and pass them to `enum_values` as well?

Reply:
@tyll can you please be more specific? I am really willing to finish this code, I already also patched ArgValidator to support number ranges for newly added ip rule parameters like sport, dport and uidrange (which are not supported on rhel/centos 7, by the way).

I am also wondering, why some tests below are pending?

Reply:
@tyll @thom311 any suggestion how to cleanly pass the list or set of valid interfaces to `enum_values` in `ArgValidatorIPRoute`? I'm not that good in Python to figure out a clean solution (which would be also accepted by maintainers...)